src/sentry/eventstore/snuba_discover/backend.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import absolute_import
+
+import six
+from copy import deepcopy
+
+from sentry.utils import snuba
+from sentry.eventstore.base import EventStorage
+from sentry.eventstore.snuba.backend import (
+    ASC_ORDERING,
+    DESC_ORDERING,
+    get_after_event_condition,
+    get_before_event_condition,
+    SnubaEventStorage,
+)
+
+
+class SnubaDiscoverEventStorage(EventStorage):
+    """
+    Experimental backend that uses the Snuba Discover dataset instead of Events
+    or Transactions directly.
+    """
+
+    def get_events(self, *args, **kwargs):
+        return SnubaEventStorage().get_events(*args, **kwargs)
+
+    def get_event_by_id(self, *args, **kwargs):
+        return SnubaEventStorage().get_event_by_id(*args, **kwargs)
```

Comment:
This would also need to go to the discover dataset long term as we don't know whether the id we have is a transaction or an event until we try to fetch it.